### PR TITLE
8340181: Shenandoah: Cleanup ShenandoahRuntime stubs

### PR DESCRIPTION
**Pull request by shipilev**
Noticed this while working on Leyden, which has to enumerate Shenandoah stubs for code archival to work.

`ShenandoahRuntime::shenandoah_clone_barrier` is excessive name. `ShenandoahRuntime::arraycopy_barrier_oop_entry` and friends is not covered by `JRT_LEAF`. This change hopefully homogenizes the namings for the stubs.

Additional testing:
 - [x] Linux x86_64 server fastdebug, `hotspot_gc_shenandoah` 
 - [x] Linux AArch64 server fastdebug, `hotspot_gc_shenandoah`

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8340181](https://bugs.openjdk.org/browse/JDK-8340181): Shenandoah: Cleanup ShenandoahRuntime stubs (**Enhancement** - P4)

### Reviewers
 * [Andrew Dinn](https://openjdk.org/census#adinn) (@adinn - **Reviewer**)
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)
 * [William Kemper](https://openjdk.org/census#wkemper) (@earthling-amzn - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21152/head:pull/21152` \
`$ git checkout pull/21152`

Update a local copy of the PR: \
`$ git checkout pull/21152` \
`$ git pull https://git.openjdk.org/jdk.git pull/21152/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21152`

View PR using the GUI difftool: \
`$ git pr show -t 21152`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21152.diff">https://git.openjdk.org/jdk/pull/21152.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21152#issuecomment-2370976177)

### CODE REVIEW

**Review by adinn — APPROVED** [review]
Thanks for cleaning this up. Looks good.

**Review by phohensee — APPROVED** [review]
In case you need a second review.

**Review by earthling-amzn — APPROVED** [review]

**Pull request closed without merging**
